include/spdlog/sinks/eventlog_sink_win32.h

```diff
@@ -0,0 +1,246 @@
+
+        // Get the required size
+        DWORD tusize = 0;
+        GetTokenInformation(current_process_token.hToken_, TokenUser, NULL, 0, &tusize);
+        if (GetLastError() != ERROR_INSUFFICIENT_BUFFER)
+            SPDLOG_THROW(win32_error("GetTokenInformation"));
+
+        std::vector<unsigned char> buffer(tusize);
+        if (!GetTokenInformation(current_process_token.hToken_, TokenUser, (LPVOID) buffer.data(), tusize, &tusize))
+            SPDLOG_THROW(win32_error("GetTokenInformation"));
+
+        return sid::duplicate((SID *) ((TOKEN_USER*) buffer.data())->User.Sid);
+    }
+};
+
+struct eventlog
+{
+    static WORD get_event_type(details::log_msg const& msg)
+    {
+        WORD type = EVENTLOG_SUCCESS;
+        if (msg.level >= level::info)
+        {
+            type = EVENTLOG_INFORMATION_TYPE;
+            if (msg.level >= level::warn)
+            {
+                type = EVENTLOG_WARNING_TYPE;
+                if (msg.level >= level::err)
+                {
+                    type = EVENTLOG_ERROR_TYPE;
+                }
+            }
+        }
+        return type;
+    }
+
+    static WORD get_event_category(details::log_msg const& msg)
+    {
+        WORD category = 1;
+        if (msg.level >= level::debug)
+        {
+            category = 2;
+            if (msg.level >= level::info)
+            {
+                category = 3;
+                if (msg.level >= level::warn)
+                {
+                    category = 4;
+                    if (msg.level >= level::err)
+                    {
+                        category = 5;
+                        if (msg.level >= level::critical)
+                        {
+                            category = 6;
+                        }
+                    }
+                }
+            }
+        }
+        return category;
+    }
+};
+
+class sink : public base_sink<spdlog::details::null_mutex>
+{
+private:
+    HANDLE hEventLog_;
+    SID* current_user_sid_;
+    std::string source_;
+    std::string log_;
+
+    void add_registry_info(std::string const& message_file_path)
+    {
+        std::string subkey("SYSTEM\\CurrentControlSet\\Services\\EventLog\\");
+        subkey += log_ + "\\" + source_;
+
+        ::HKEY hkey {};
+        DWORD disposition {};
+        long stat = RegCreateKeyEx(HKEY_LOCAL_MACHINE, subkey.c_str(), 0, NULL,
+                                   REG_OPTION_NON_VOLATILE, KEY_SET_VALUE, NULL,
+                                   &hkey, &disposition);
+        if (stat == ERROR_SUCCESS)
+        {
+            if (disposition == REG_CREATED_NEW_KEY && !message_file_path.empty())
+            {
+                auto const expanded_message_file_path_length = ExpandEnvironmentStrings(message_file_path.c_str(), (LPSTR) &disposition, 0);
+                std::vector<char>expanded_message_file_path(expanded_message_file_path_length + 1);
```

Comment:
I get warning here: Severity	Code	Description	Project	File	Line
Warning	C26451	Arithmetic overflow: Using operator '+' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '+' to avoid overflow